11_FFTBloom/app_resources/fft_convolve_ifft.hlsl

```diff
@@ -0,0 +1,260 @@
+#include "common.hlsl"
+#include "nbl/builtin/hlsl/workgroup/fft.hlsl"
+
+// TODO: There's a lot of redundant stuff in every FFT file, I'd like to move that to another file that I can sourceFmt at runtime then include in all of them (something like 
+// a runtime common.hlsl)
+
+/*
+ * Remember we have these defines:
+ * _NBL_HLSL_WORKGROUP_SIZE_
+ * ELEMENTS_PER_THREAD
+ * (may be defined) USE_HALF_PRECISION
+ * KERNEL_SCALE
+*/
+
+#ifdef USE_HALF_PRECISION
+#define scalar_t float16_t
+#define FORMAT rg16f
+#else
+#define scalar_t float32_t
+#define FORMAT rg32f
+#endif
```

Comment:
look at this magic then https://github.com/Devsh-Graphics-Programming/Nabla/blob/e1a87e757995a1d531d5517c9a23e6d07b9dbafe/src/nbl/video/utilities/CComputeBlit.cpp#L65